tests/test_maintel_m1m3_check_actuators.py

```diff
@@ -0,0 +1,142 @@
+            side_effect=self.mock_test_bump
+        )
+        self.script.mtcs.enter_m1m3_engineering_mode = unittest.mock.AsyncMock()
+        self.script.mtcs.exit_m1m3_engineering_mode = unittest.mock.AsyncMock()
+        self.script.mtcs.assert_liveliness = unittest.mock.AsyncMock()
+        self.script.mtcs.assert_all_enabled = unittest.mock.AsyncMock()
+        self.script.mtcs.assert_m1m3_detailed_state = unittest.mock.AsyncMock()
+        self.script.mtcs.assert_feasibility = unittest.mock.AsyncMock()
+
+        self.bump_test_status = types.SimpleNamespace(
+            testState=[BumpTest.NOTTESTED] * len(self.m1m3_actuator_ids)
+        )
+
+        return (self.script,)
+
+    async def mock_test_bump(self, actuator_id):
+        await asyncio.sleep(5.0)
+        self.bump_test_status.testState[actuator_id] = BumpTest.PASSED
+
+    async def test_configure_good(self):
+        """Testing a valid configuration"""
+
+        # Configure with "all" actuators
+        async with self.make_script():
+            actuators = "all"
+
+            await self.configure_script(
+                actuators=actuators
+            )
+
+            assert self.script.actuators == self.m1m3_actuator_ids
+
+        # Try configure with a list of valid actuators ids
+        async with self.make_script():
+            actuators = [101, 210, 301, 410]
+
+            await self.configure_script(
+                actuators=actuators,
+            )
+
+            assert self.script.actuators == actuators
+
+    async def test_configure_bad(self):
+        """Testing an invalid configuration"""
+
+        async with self.make_script():
+            # Invalid actuators: 501 and 505
+            actuators = [501, 505]
+
+            # If actuators_bad_ids is not empty, it should raise a ValueError
+            actuators_bad_ids = [
+                actuator
+                for actuator in actuators
+                if actuator not in self.m1m3_actuator_ids
+            ]
+            if actuators_bad_ids:
+                with self.assertRaises(ValueError):
+                    await self.configure_script(
+                        actuators=actuators_bad_ids,
+                    )
+
+    async def test_run(self):
+        # Run the script
+        async with self.make_script():
+            # Assert feasibility
+            self.script.mtcs.assert_feasibility()
+
+            # Await once for all mock methods defined on the basic_make_script
+            self.script.mtcs.enter_m1m3_engineering_mode.assert_awaited_once()
+            self.script.mtcs.exit_m1m3_engineering_mode.assert_awaited_once()
+            self.script.mtcs.assert_liveliness.assert_awaited_once()
+            self.script.mtcs.assert_all_enabled.assert_awaited_once()
+            self.script.mtcs.assert_m1m3_detailed_state.assert_awaited_once()
+            self.script.mtcs.assert_feasibility.assert_awaited_once()
```

Comment:
This doesn't exist. You might be thinking of mocking `self.script.assert_feasibility`, but you shouldn't. Mock only external stuff (methods from mtcs).